treeherder/client/thclient/client.py

```diff
@@ -678,17 +675,36 @@ def __init__(
         self.protocol = protocol
         self.timeout = timeout
 
-        # ensure the required parameters are given
-        if not self.project:
-            msg = "{0}: project required for posting".format(
-                self.__class__.__name__)
-            raise TreeherderClientError(msg, [])
+    def _get_uri(self, project, endpoint, data, method='POST'):
+
+        uri = '{0}://{1}/api/project/{2}/{3}/'.format(
+            self.protocol, self.host, project, endpoint
+            )
 
-    def post(self, collection_inst):
-        """Shortcut method to send a treeherder collection via POST
+        if self.use_oauth:
+            oauth_client = OauthClient(self.oauth_key, self.oauth_secret,
+                                       project)
+            uri = oauth_client.get_signed_uri(data, uri, method)
+
+        return uri
 
+    def _post_json(self, project, endpoint, jsondata):
```

Comment:
In the spirit of making further modifications easier, I would add timeout as a keyword argument default to None. In this way the user can decide to pass a different timeout based on the content posted/endpoint used. And if that is None, the client timeout will be used